File: WorldFeed/Server/src/WorldFeed.Common/Public/Parameters/ListsClient/CreateListParameters.cs
namespace WorldFeed.Common.Public.Parameters.ListsClient
{
    using WorldFeed.Common.Public.Parameters.ListsClient.Interfaces;

    /// <summary>
    /// For more information visit : https://developer.twitter.com/en/docs/accounts-and-users/create-manage-lists/api-reference/post-lists-create
    /// </summary>
    /// <inheritdoc />
    public interface ICreateListParameters : IListMetadataParameters
    {
    }

    /// <inheritdoc />
    public class CreateListParameters : ListMetadataParameters, ICreateListParameters
    {
        public CreateListParameters(string name)
        {
            Name = name;
        }

        public CreateListParameters(IListMetadataParameters parameters) 
            : base(parameters)
        {
        }
    }
}

File: Sharebook/Client/src/app/core/Public/Parameters/ListsClient/CreateListParameters.ts
import {IListMetadataParameters, ListMetadataParameters} from "./IListMetadataParameters";

// For more information visit : https://developer.twitter.com/en/docs/accounts-and-users/create-manage-lists/api-reference/post-lists-create

export interface ICreateListParameters extends IListMetadataParameters {
}

export class CreateListParameters extends ListMetadataParameters implements ICreateListParameters {
  constructor(nameOrParameters: string | IListMetadataParameters) {
    if (typeof nameOrParameters === 'string') {
      super();
      super.name = nameOrParameters;
    } else {
      super(nameOrParameters);
    }
  }
}

// public CreateListParameters(string name)
// {
//   Name = name;
// }
//
// public CreateListParameters(IListMetadataParameters parameters) : base(parameters)
// {
// }
